From b717287776c83e4da35f93cb2cf623276ed2c10e Mon Sep 17 00:00:00 2001
From: Jeff Xu <jeffxu@google.com>
Date: Sat, 22 Oct 2022 16:38:44 +0000
Subject: [PATCH] CHROMIUM: ChromeOS LSM: add check for BPF_SYSCALL

Add compile time check for CONFIG_BPF_SYSCALL, in case BPF_SYSCALL
is not defined in build. This is to be consistent as in 5.10.

BUG=b:251196147
TEST=CQ

Signed-off-by: Jeff Xu <jeffxu@google.com>
Change-Id: I7f3781e429f42567f41a60c95d8842401784e442
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/3972729
Reviewed-by: Aashay Shringarpure <aashay@google.com>

[rebase61(tzungbi):
    Squashed:
	FIXUP: CHROMIUM: ChromeOS LSM: add check for BPF_SYSCALL
]
Signed-off-by: Tzung-Bi Shih <tzungbi@chromium.org>
---

Notes:
    73106dc3647a0a14adac7ef40c9665b7d0ac4554
    05dacc66cffd625367704a17aabd6b6a75c24979

 security/chromiumos/lsm.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/security/chromiumos/lsm.c b/security/chromiumos/lsm.c
index ce165e6f7047..b348acf36873 100644
--- a/security/chromiumos/lsm.c
+++ b/security/chromiumos/lsm.c
@@ -41,8 +41,6 @@
 #include "inode_mark.h"
 #include "utils.h"
 
-static const char secagentd[] = "/usr/sbin/secagentd";
-
 #if defined(CONFIG_SECURITY_CHROMIUMOS_NO_UNPRIVILEGED_UNSAFE_MOUNTS) || \
 	defined(CONFIG_SECURITY_CHROMIUMOS_NO_SYMLINK_MOUNT)
 static void report(const char *origin, const struct path *path, char *operation)
@@ -275,6 +273,10 @@ static int chromiumos_locked_down(enum lockdown_reason what)
 	return 0;
 }
 
+#ifdef CONFIG_BPF_SYSCALL
+
+static const char secagentd[] = "/usr/sbin/secagentd";
+
 static int chromiumos_bpf(int cmd, union bpf_attr *attr, unsigned int size)
 {
 	char buf[128];
@@ -295,6 +297,7 @@ static int chromiumos_bpf(int cmd, union bpf_attr *attr, unsigned int size)
 
 	return 0;
 }
+#endif
 
 static struct security_hook_list chromiumos_security_hooks[] = {
 	LSM_HOOK_INIT(sb_mount, chromiumos_security_sb_mount),
@@ -303,7 +306,9 @@ static struct security_hook_list chromiumos_security_hooks[] = {
 	LSM_HOOK_INIT(sb_eat_lsm_opts, chromiumos_sb_eat_lsm_opts),
 	LSM_HOOK_INIT(bprm_creds_for_exec, chromiumos_bprm_creds_for_exec),
 	LSM_HOOK_INIT(locked_down, chromiumos_locked_down),
+#ifdef CONFIG_BPF_SYSCALL
 	LSM_HOOK_INIT(bpf, chromiumos_bpf),
+#endif
 };
 
 static int __init chromiumos_security_init(void)
-- 
2.38.1.584.g0f3c55d4c2-goog

